In [ ]:
import pandas as pd

In [ ]:
monthly_expenses_sheet_id = '10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw'
monthly_expenses = pd.read_excel(f'https://docs.google.com/spreadsheets/d/{monthly_expenses_sheet_id}/export?format=xlsx', sheet_name='Sheet1')
monthly_expenses

#      month           category                  item  amount
0    1 2023-01-01       Server Costs           AWS Hosting    8000
1    2 2023-01-01       Server Costs  Google Cloud Storage    4000
2    3 2023-01-01  Software Licenses        Atlassian Jira    1200
3    4 2023-01-01  Software Licenses                 Slack     800
4    5 2023-01-01  Software Licenses            Salesforce    1500
5    6 2023-01-01  Software Licenses                  Zoom     500
6    7 2023-01-01      Office Rental           Office Rent   10000
7    8 2023-01-01              Other       Office Supplies     500
8    9 2023-01-01              Other       Travel Expenses    3000
9   10 2023-02-01       Server Costs           AWS Hosting    8200
10  11 2023-02-01       Server Costs  Google Cloud Storage    4200
11  12 2023-02-01  Software Licenses        Atlassian Jira    1300
12  13 2023-02-01  Software Licenses                 Slack     850
13  14 2023-02-01  Software Licenses            Salesforce    1600
14  15 2023-02-01  Software Licenses                  Zoom     520
15  16 2023-02-01      Office Rental           Office Rent   10000
16  17 2023-02-01              Other       Office Supplies     550
17  18 2023-02-01              Other       Travel Expenses    3100
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700
23  24 2023-03-01  Software Licenses                  Zoom     540
24  25 2023-03-01      Office Rental           Office Rent   10000
25  26 2023-03-01              Other       Office Supplies     600
26  27 2023-03-01              Other       Travel Expenses    3200

In [ ]:
last_month_expense = monthly_expenses[monthly_expenses['month'] == monthly_expenses['month'].max()]
last_month_expense

#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700
23  24 2023-03-01  Software Licenses                  Zoom     540
24  25 2023-03-01      Office Rental           Office Rent   10000
25  26 2023-03-01              Other       Office Supplies     600
26  27 2023-03-01              Other       Travel Expenses    3200

In [ ]:
crm_expense = last_month_expense[last_month_expense['item'] == 'Salesforce']['amount'].values[0]
crm_expense

np.int64(1700)

In [ ]:
TotalExpense = crm_expense

In [ ]:
payroll_sheet_id = '1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4'
payroll_monthly = pd.read_excel(f'https://docs.google.com/spreadsheets/d/{payroll_sheet_id}/export?format=xlsx', sheet_name='Sheet1')
payroll_monthly

month   department   employee_name            position  paid
0  2023-01-01        Sales        John Doe       Sales Manager  1500
1  2023-01-01        Sales      Jane Smith     Sales Associate   600
2  2023-01-01        Sales       Jim Brown     Sales Associate   700
3  2023-01-01        Sales    Laura Miller     Sales Associate   800
4  2023-01-01    Marketing   Alice Johnson   Marketing Manager  1650
5  2023-01-01    Marketing       Bob Davis  Content Specialist   700
6  2023-01-01  Engineering  Charlie Wilson    Senior Developer  2500
7  2023-01-01  Engineering       Diana Lee           Developer  1700
8  2023-01-01  Engineering       Eva White    Junior Developer  1000
9  2023-01-01           HR     Frank Green          HR Manager  1300
10 2023-01-01           HR      Gary Black       HR Specialist   700
11 2023-01-01      Support    Hannah Scott        Support Lead   600
12 2023-01-01      Support      Ian Harris  Support Specialist   500
13 2023-01-01      Support     Emily Clark  Support Specialist   450
14 2023-01-01    Analytics   Michael Brown   Analytics Manager  1700
15 2023-01-01    Analytics    Sarah Wilson        Data Analyst  1200
16 2023-01-01    Analytics       David Lee        Data Analyst  1300
17 2023-02-01        Sales        John Doe       Sales Manager  1500
18 2023-02-01        Sales      Jane Smith     Sales Associate   600
19 2023-02-01        Sales       Jim Brown     Sales Associate   700
20 2023-02-01        Sales    Laura Miller     Sales Associate  5100
21 2023-02-01    Marketing   Alice Johnson   Marketing Manager  1650
22 2023-02-01    Marketing       Bob Davis  Content Specialist   700
23 2023-02-01  Engineering  Charlie Wilson    Senior Developer  2500
24 2023-02-01  Engineering       Diana Lee           Developer  1700
25 2023-02-01  Engineering       Eva White    Junior Developer  1000
26 2023-02-01           HR     Frank Green          HR Manager  1300
27 2023-02-01           HR      Gary Black       HR Specialist   700
28 2023-02-01      Support    Hannah Scott        Support Lead   600
29 2023-02-01      Support      Ian Harris  Support Specialist   500
30 2023-02-01      Support     Emily Clark  Support Specialist   450
31 2023-02-01    Analytics   Michael Brown   Analytics Manager  1700
32 2023-02-01    Analytics    Sarah Wilson        Data Analyst  1200
33 2023-02-01    Analytics       David Lee        Data Analyst  1300
34 2023-03-01        Sales        John Doe       Sales Manager  1500
35 2023-03-01        Sales      Jane Smith     Sales Associate   600
36 2023-03-01        Sales       Jim Brown     Sales Associate   700
37 2023-03-01        Sales    Laura Miller     Sales Associate   800
38 2023-03-01    Marketing   Alice Johnson   Marketing Manager  1650
39 2023-03-01    Marketing       Bob Davis  Content Specialist   700
40 2023-03-01  Engineering  Charlie Wilson    Senior Developer  2500
41 2023-03-01  Engineering       Diana Lee           Developer  1700
42 2023-03-01  Engineering       Eva White    Junior Developer  1000
43 2023-03-01           HR     Frank Green          HR Manager  1300
44 2023-03-01           HR      Gary Black       HR Specialist   700
45 2023-03-01      Support    Hannah Scott        Support Lead   600
46 2023-03-01      Support      Ian Harris  Support Specialist   500
47 2023-03-01      Support     Emily Clark  Support Specialist   450
48 2023-03-01    Analytics   Michael Brown   Analytics Manager  1700
49 2023-03-01    Analytics    Sarah Wilson        Data Analyst  1200
50 2023-03-01    Analytics       David Lee        Data Analyst  1300

In [ ]:
last_month_payroll = payroll_monthly[payroll_monthly['month'] == payroll_monthly['month'].max()]
last_month_payroll

month   department   employee_name            position  paid
34 2023-03-01        Sales        John Doe       Sales Manager  1500
35 2023-03-01        Sales      Jane Smith     Sales Associate   600
36 2023-03-01        Sales       Jim Brown     Sales Associate   700
37 2023-03-01        Sales    Laura Miller     Sales Associate   800
38 2023-03-01    Marketing   Alice Johnson   Marketing Manager  1650
39 2023-03-01    Marketing       Bob Davis  Content Specialist   700
40 2023-03-01  Engineering  Charlie Wilson    Senior Developer  2500
41 2023-03-01  Engineering       Diana Lee           Developer  1700
42 2023-03-01  Engineering       Eva White    Junior Developer  1000
43 2023-03-01           HR     Frank Green          HR Manager  1300
44 2023-03-01           HR      Gary Black       HR Specialist   700
45 2023-03-01      Support    Hannah Scott        Support Lead   600
46 2023-03-01      Support      Ian Harris  Support Specialist   500
47 2023-03-01      Support     Emily Clark  Support Specialist   450
48 2023-03-01    Analytics   Michael Brown   Analytics Manager  1700
49 2023-03-01    Analytics    Sarah Wilson        Data Analyst  1200
50 2023-03-01    Analytics       David Lee        Data Analyst  1300

In [ ]:
last_month_payroll_sale_mkt_expense = last_month_payroll[last_month_payroll['department'].isin(['Sales', 'Marketing'])]['paid'].sum()
last_month_payroll_sale_mkt_expense

np.int64(5950)

In [ ]:
TotalExpense += last_month_payroll_sale_mkt_expense

In [ ]:
marketing_spendings_sheet_id = '1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c'
marketing_spendings = pd.read_excel(f'https://docs.google.com/spreadsheets/d/{marketing_spendings_sheet_id}/export?format=xlsx', sheet_name='Sheet1')
marketing_spendings

date       channel  spending
0   2023-01-01    Google Ads       784
1   2023-01-01  Facebook Ads       659
2   2023-01-01  LinkedIn Ads       729
3   2023-01-01   Twitter Ads       292
4   2023-01-02    Google Ads       935
..         ...           ...       ...
355 2023-03-30   Twitter Ads       960
356 2023-03-31    Google Ads       143
357 2023-03-31  Facebook Ads       183
358 2023-03-31  LinkedIn Ads       533
359 2023-03-31   Twitter Ads       909

[360 rows x 3 columns]

In [ ]:
last_month_marketing_expense = marketing_spendings[marketing_spendings['date'].dt.month == marketing_spendings['date'].max().month]['spending'].sum()
last_month_marketing_expense

np.int64(68830)

In [ ]:
TotalExpense += last_month_marketing_expense

In [ ]:
TotalExpense

np.int64(76480)

In [ ]:
receipt_history_sheet_id = '1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE'
receipt_history = pd.read_excel(f'https://docs.google.com/spreadsheets/d/{receipt_history_sheet_id}/export?format=xlsx', sheet_name='Sheet1')
receipt_history

date  customer_id  receipt_amount  new_customer
0   2023-01-01         2653              67             1
1   2023-01-01         2731             271             1
2   2023-01-01         1277             231             0
3   2023-01-01         2094             107             0
4   2023-01-01         1314             416             0
..         ...          ...             ...           ...
924 2023-03-31         2079             358             0
925 2023-03-31         2089             375             0
926 2023-03-31         1355             249             0
927 2023-03-31         1062             111             0
928 2023-03-31         1590             271             0

[929 rows x 4 columns]

In [ ]:
last_month_new_customer = receipt_history[(receipt_history['date'].dt.month == receipt_history['date'].max().month) & (receipt_history['new_customer'] == 1)]
last_month_new_customer

date  customer_id  receipt_amount  new_customer
621 2023-03-01         2406             426             1
622 2023-03-01         2761              41             1
625 2023-03-01         2844             252             1
627 2023-03-01         2679             323             1
633 2023-03-02         1475              40             1
..         ...          ...             ...           ...
912 2023-03-29         1746             262             1
913 2023-03-29         2269             484             1
914 2023-03-29         2935             478             1
919 2023-03-29         1297             496             1
920 2023-03-30         2151             482             1

[63 rows x 4 columns]

In [ ]:
number_customer = len(last_month_new_customer)
number_customer

63

#1. CAC: Customer Acquisition Cost

In [ ]:
CAC = TotalExpense/number_customer
CAC

np.float64(1213.968253968254)

#2. ARPU: Average Revenue Per User

In [ ]:
receipt_history

date  customer_id  receipt_amount  new_customer
0   2023-01-01         2653              67             1
1   2023-01-01         2731             271             1
2   2023-01-01         1277             231             0
3   2023-01-01         2094             107             0
4   2023-01-01         1314             416             0
..         ...          ...             ...           ...
924 2023-03-31         2079             358             0
925 2023-03-31         2089             375             0
926 2023-03-31         1355             249             0
927 2023-03-31         1062             111             0
928 2023-03-31         1590             271             0

[929 rows x 4 columns]

In [ ]:
total_revenue = receipt_history[(receipt_history['date'].dt.month == receipt_history['date'].max().month)]['receipt_amount'].sum()
total_revenue

np.int64(83033)

In [ ]:
number_customer_last_month = receipt_history[(receipt_history['date'].dt.month == receipt_history['date'].max().month)]['customer_id'].nunique()
number_customer_last_month

292

In [ ]:
ARPU = total_revenue/number_customer_last_month
ARPU

np.float64(284.3595890410959)

#3.COGS: Cost of Goods Sold

In [ ]:
monthly_expenses

#      month           category                  item  amount
0    1 2023-01-01       Server Costs           AWS Hosting    8000
1    2 2023-01-01       Server Costs  Google Cloud Storage    4000
2    3 2023-01-01  Software Licenses        Atlassian Jira    1200
3    4 2023-01-01  Software Licenses                 Slack     800
4    5 2023-01-01  Software Licenses            Salesforce    1500
5    6 2023-01-01  Software Licenses                  Zoom     500
6    7 2023-01-01      Office Rental           Office Rent   10000
7    8 2023-01-01              Other       Office Supplies     500
8    9 2023-01-01              Other       Travel Expenses    3000
9   10 2023-02-01       Server Costs           AWS Hosting    8200
10  11 2023-02-01       Server Costs  Google Cloud Storage    4200
11  12 2023-02-01  Software Licenses        Atlassian Jira    1300
12  13 2023-02-01  Software Licenses                 Slack     850
13  14 2023-02-01  Software Licenses            Salesforce    1600
14  15 2023-02-01  Software Licenses                  Zoom     520
15  16 2023-02-01      Office Rental           Office Rent   10000
16  17 2023-02-01              Other       Office Supplies     550
17  18 2023-02-01              Other       Travel Expenses    3100
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700
23  24 2023-03-01  Software Licenses                  Zoom     540
24  25 2023-03-01      Office Rental           Office Rent   10000
25  26 2023-03-01              Other       Office Supplies     600
26  27 2023-03-01              Other       Travel Expenses    3200

In [ ]:
production_item = ['AWS Hosting', 'Google Cloud Storage', 'Atlassian Jira']
last_month_production_expense = last_month_expense[last_month_expense['item'].isin(production_item)]['amount'].sum()
last_month_production_expense

np.int64(14200)

In [ ]:
shared_software_item = ['Zoom', 'Slack']
last_month_share_software_item = last_month_expense[last_month_expense['item'].isin(shared_software_item)]['amount'].sum()
last_month_share_software_item

np.int64(1440)

In [ ]:
server_n_software_exprense = last_month_production_expense + last_month_share_software_item * 0.6
server_n_software_exprense

np.float64(15064.0)

In [ ]:
last_month_production_salary = last_month_payroll[last_month_payroll['department']=='Engineering']['paid'].sum()
last_month_production_salary

np.int64(5200)

In [ ]:
COGS = server_n_software_exprense + last_month_production_salary
COGS

np.float64(20264.0)

#4.Gross Margin

In [ ]:
GrossMargin = (total_revenue - COGS)/total_revenue * 100
GrossMargin

np.float64(75.5952452639312)

#5.LTV: Customer Lifetime Value

In [ ]:
life_sheet_id = '1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI'
life_data = pd.read_excel(f'https://docs.google.com/spreadsheets/d/{life_sheet_id}/export?format=xlsx', sheet_name='Sheet1')
life_data

Unnamed: 0 start_date churn_date
0         1000 2021-11-15 2022-09-14
1         1001 2022-04-15 2023-02-16
2         1002 2022-10-30 2023-02-04
3         1003 2021-08-22 2023-02-07
4         1004 2021-08-23 2022-02-02
..         ...        ...        ...
95        1095 2023-01-09 2023-02-28
96        1096 2020-07-08 2020-08-09
97        1097 2020-10-18 2022-05-08
98        1098 2020-06-17 2022-06-13
99        1099 2020-06-25 2023-02-12

[100 rows x 3 columns]

In [ ]:
life_data['lifespan_day'] = (life_data['churn_date'] - life_data['start_date']).dt.days
life_data

Unnamed: 0 start_date churn_date  lifespan_day
0         1000 2021-11-15 2022-09-14           303
1         1001 2022-04-15 2023-02-16           307
2         1002 2022-10-30 2023-02-04            97
3         1003 2021-08-22 2023-02-07           534
4         1004 2021-08-23 2022-02-02           163
..         ...        ...        ...           ...
95        1095 2023-01-09 2023-02-28            50
96        1096 2020-07-08 2020-08-09            32
97        1097 2020-10-18 2022-05-08           567
98        1098 2020-06-17 2022-06-13           726
99        1099 2020-06-25 2023-02-12           962

[100 rows x 4 columns]

In [ ]:
avg_lifespan_days = life_data['lifespan_day'].mean()
avg_lifespan_month = avg_lifespan_days / 30

In [ ]:
LTV = ARPU * avg_lifespan_month * (GrossMargin/100)
LTV

np.float64(2115.515931506849)

#6.LTV / CAC

In [ ]:
LTV_CAC = LTV / CAC
LTV_CAC

np.float64(1.742645184164899)

#7. Conclusion

#Kết luận
- Tỷ lệ LTV/CAC = 1.74 cho thấy mỗi đô la chi ra để thu hút khách hàng mang về 1.74 đô doanh thu trọn đời từ khách đó -> Mặc dù có lời, nhưng tỷ lệ này vẫn dưới mức lý tưởng là 3.0 – tức doanh nghiệp vẫn chưa khai thác được tối đa tiềm năng lợi nhuận từ mỗi khách hàng.

- Gross Margin = 75.6% là một con số rất tích cực, cho thấy doanh nghiệp giữ lại phần lớn doanh thu sau khi trừ chi phí trực tiếp (COGS).

- ARPU = 284.36 USD tương đối thấp so với LTV = 2,115.52 USD, điều này cho thấy khách hàng có xu hướng gắn bó lâu dài hoặc mua nhiều lần (LTV cao hơn ARPU nhiều lần).

#Khuyến nghị
 1. Tối ưu hóa chi phí thu hút khách hàng (CAC)
CAC hiện tại trên 1,200 USD là khá cao → cần tìm cách giảm xuống dưới 1,000 USD để tăng tỷ lệ LTV/CAC.

- Các cách giảm CAC:

   - Tăng tỷ lệ chuyển đổi trong marketing.

   - Tận dụng kênh truyền miệng hoặc giới thiệu bạn bè.

   - Đầu tư vào SEO & nội dung chất lượng thay vì chỉ quảng cáo trả phí.

2. Tăng LTV thông qua giữ chân khách hàng
- Tăng tần suất mua hàng bằng email marketing, ưu đãi định kỳ.

- Phát triển chương trình chăm sóc khách hàng trung thành (loyalty program).

- Cải thiện chất lượng sản phẩm & trải nghiệm dịch vụ để kéo dài vòng đời khách hàng.

3. Theo dõi chặt chẽ chỉ số LTV:CAC
- Mục tiêu nên là LTV:CAC > 3.0

- Đặt KPI cụ thể cho từng quý để cải thiện dần dần.